open-machine-learning-jupyter-book/ml-advanced/clustering/introduction-to-clustering.md

```diff
@@ -0,0 +1,323 @@
+
+Is there any convergence in this dataset around a song's perceived popularity and danceability? A FacetGrid shows that there are concentric circles that line up, regardless of genre. Could it be that Nigerian tastes converge at a certain level of danceability for this genre?  
+
+✅ Try different datapoints (energy, loudness, speechiness) and more or different musical genres. What can you discover? Take a look at the `df.describe()` table to see the general spread of the data points.
+
+### Exercise - data distribution
+
+Are these three genres significantly different in the perception of their danceability, based on their popularity?
+
+1. Examine our top three genres data distribution for popularity and danceability along a given x and y axis.
+
+    ```python
+    sns.set_theme(style="ticks")
+    
+    g = sns.jointplot(
+        data=df,
+        x="popularity", y="danceability", hue="artist_top_genre",
+        kind="kde",
+    )
+    ```
+
+    You can discover concentric circles around a general point of convergence, showing the distribution of points.
+
+    > 🎓 Note that this example uses a KDE (Kernel Density Estimate) graph that represents the data using a continuous probability density curve. This allows us to interpret data when working with multiple distributions.
+
+    In general, the three genres align loosely in terms of their popularity and danceability. Determining clusters in this loosely-aligned data will be a challenge:
+
+    ![distribution](../../../images/clustering/distribution.png)
+
+1. Create a scatter plot:
+
+    ```python
+    sns.FacetGrid(df, hue="artist_top_genre", size=5) \
+       .map(plt.scatter, "popularity", "danceability") \
+       .add_legend()
+    ```
+
+    A scatterplot of the same axes shows a similar pattern of convergence
+
+    ![Facetgrid](../../../images/clustering/facetgrid.png)
+
+In general, for clustering, you can use scatterplots to show clusters of data, so mastering this type of visualization is very useful. In the next lesson, we will take this filtered data and use k-means clustering to discover groups in this data that see to overlap in interesting ways.
+
+---
+
+## 🚀Challenge
+
+In preparation for the next lesson, make a chart about the various clustering algorithms you might discover and use in a production environment. What kinds of problems is the clustering trying to address?
+
+## [Post-lecture quiz](https://gray-sand-07a10f403.1.azurestaticapps.net/quiz/28/)
+
+## Review & Self Study
+
+Before you apply clustering algorithms, as we have learned, it's a good idea to understand the nature of your dataset. Read more on this topic [here](https://www.kdnuggets.com/2019/10/right-clustering-algorithm.html)
+
+[This helpful article](https://www.freecodecamp.org/news/8-clustering-algorithms-in-machine-learning-that-all-data-scientists-should-know/) walks you through the different ways that various clustering algorithms behave, given different data shapes.
+
+## Assignment
```

Comment:
We call this as Your turn section. https://open-academy.github.io/machine-learning/data-science/working-with-data/relational-databases.html#your-turn